samples/boards/nrf/dynamic_pinctrl/prj.conf

```diff
@@ -0,0 +1,7 @@
+CONFIG_PINCTRL_DYNAMIC=y
+# Make sure GPIO is initialized before UART. This is required because we need
+# to read a GPIO pin to decide if UART pins have to be remapped or not, an
+# operation that needs to happen before UART device is initialized.
+CONFIG_GPIO_INIT_PRIORITY=50
+CONFIG_SERIAL_INIT_PRIORITY=70
+CONFIG_CONSOLE_INIT_PRIORITY=80
```

Comment:
What about adding here also:
```
CONFIG_REMAP_INIT_PRIORITY=60
```

Reply:
I've refactored this a little bit and added build assertion, please check again

Reply:
Yes, the build assertion is a good idea. And everything in this file is clear now. Thanks.